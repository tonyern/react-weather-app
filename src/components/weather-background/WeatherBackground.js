import React, { useState } from "react";
import "./weather-background.css";
import SearchBar from "../search-bar/SearchBar";
import WeatherInfo from "../weather-display/WeatherInfo";

const WeatherBackground = () => {
  const [weather, setWeather] = useState({});

  const changeBackground = () => {
    let background = "App ";

    if (typeof weather.main != "undefined") {
      background = background.concat(weather.weather[0].main);
    }

    console.log(background);

    return background;
  };

  return (
    <div className={changeBackground()}>
      <SearchBar searchProps={setWeather} />
      <WeatherInfo weatherData={weather} />
    </div>
  );
};

export default WeatherBackground;
