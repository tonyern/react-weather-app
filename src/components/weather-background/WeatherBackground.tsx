import React, { useState } from "react";
import "./weather-background.css";
import SearchBar from "../search-bar/SearchBar";
import WeatherInfo from "../weather-display/WeatherInfo";

const WeatherBackground = (): JSX.Element => {
  const [weather, setWeather] = useState({});

  const changeBackground = (): string => {
    let background = "App ";

    // @ts-ignore
    if (typeof weather.main != "undefined") {
      // @ts-ignore
      background = background.concat(weather.weather[0].main);
    }

    return background;
  };

  return (
    <div className={changeBackground()} data-testid="weather-background-test">
      <main>
        <SearchBar searchProps={setWeather} />
        <WeatherInfo weatherData={weather} />
      </main>
    </div>
  );
};

export default WeatherBackground;
