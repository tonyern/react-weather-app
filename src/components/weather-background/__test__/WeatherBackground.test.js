import React from "react";
import ReactDOM from "react-dom";
import render, { fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { renderHook } from "@testing-library/react-hooks";
import "@testing-library/jest-dom/extend-expect";
import WeatherBackground from "../WeatherBackground";

it("Weather background renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<WeatherBackground />, div);
});

it("Weather background snapshot test", () => {
  const tree = renderer.create(<WeatherBackground />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Testing calls to changeBackground function", () => {
  const div = document.createElement("div");
  const result = ReactDOM.render(<WeatherBackground />, div);

  const changeBackground = jest.fn();
});

it("Making sure weather hook exist with no problem", () => {
  const { result } = renderHook(() => WeatherBackground());
  expect(result.current.weather).toBe();
});
