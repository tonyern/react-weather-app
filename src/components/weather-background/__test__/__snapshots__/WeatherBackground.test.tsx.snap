// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather background snapshot test 1`] = `
<div
  className="App"
  data-testid="weather-background-test"
>
  <main>
    <div
      className="search-box"
    >
      <input
        className="search-bar"
        data-testid="search-input-test"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search City"
        type="text"
        value=""
      />
    </div>
    <div
      data-testid="weather-info-test"
    >
      <div
        className="weather-box"
      >
        <div
          className="weather-detail"
        >
          Please Enter a City to get Weather Info
        </div>
      </div>
    </div>
  </main>
</div>
`;
